### PR TITLE
Handle emails with different 'Message-Id' casing

### PR DESCRIPTION
**Pull request by wagnerand**
Fixes https://github.com/mozilla/addons/issues/361.

### CODE REVIEW

File: src/olympia/activity/tasks.py [hunk 1]
```diff
@@ -17,7 +18,7 @@ def process_email(message, **kwargs):
     if not msg_id:
         custom_headers = message.get('CustomHeaders', [])
         for header in custom_headers:
-            if header.get('Name') == 'Message-ID':
+            if re.match(r'^Message-I[Dd]$', header.get('Name')):
```
Comment (diox) [inline src/olympia/activity/tasks.py]: How about just `header.get('Name', '').lower() == 'message-id` ?

Comment (wagnerand) [inline src/olympia/activity/tasks.py]: I had that initially. 😞

Comment (diox) [inline src/olympia/activity/tasks.py]: When in doubt, avoid regexps :)

Comment (wagnerand) [inline src/olympia/activity/tasks.py]: Alright! Done.

**Pull request merged**